### PR TITLE
Improvements to the Secrets Management Component

### PR DESCRIPTION
**Pull request by ThaminduDilshan**
### Proposed changes in this pull request
Changes to the secret management API.
- Secrets type API will be removed and secret types will be stored in the framework.
- Secrets API definition will be changed.
- secret type `ADAPTIVE_AUTH_CALL_CHOREO` will be renamed to `ADAPTIVE_AUTH`.
- Unit tests.

## Related Issues
- https://github.com/wso2/product-is/issues/12643

### When should this PR be merged
- After the relevant changes are done in Frontend.
- Merging the related PRs in the correct order.

### Follow up actions
- Database migration.
- Merge related PRs.

### Migrations (if applicable)
- Will follow a database migration task (dropping a table, dropping some foreign key constraints, update secret type stored values).

### Related PRs.
- https://github.com/wso2/identity-api-server/pull/326
- https://github.com/wso2/product-is/pull/12644
- https://github.com/wso2-extensions/identity-conditional-auth-functions/pull/86
- https://github.com/wso2/product-is/pull/12655
- https://github.com/wso2/carbon-utils/pull/23

### CODE REVIEW

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/internal/FrameworkServiceComponent.java [hunk 1]
```diff
@@ -137,6 +139,7 @@
     private static final String LOGIN_CONTEXT_SERVLET_URL = "/logincontext";
     private static final String LONGWAITSTATUS_SERVLET_URL = "/longwaitstatus";
     private static final Log log = LogFactory.getLog(FrameworkServiceComponent.class);
+    private static final String SECRET_TYPE = "adaptive-auth";
```
Comment (ThaminduDilshan) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/internal/FrameworkServiceComponent.java:142]: Change to `SECRET_TYPE_ADAPTIVE_AUTH`

Comment (ThaminduDilshan) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/internal/FrameworkServiceComponent.java]: Fixed

File: components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManagerImpl.java [hunk 2]
```diff
@@ -87,261 +79,182 @@ public Secret addSecret(String secretTypeName, Secret secret) throws SecretManag
...
-            log.debug(secret.getSecretName() + " secret replaced successfully.");
+            log.debug("Secret: " + secret.getSecretId() + " is retrieved successfully.");
         }
         return secret;
     }
 
     @Override
-    public Secret updateSecretValue(String secretTypeName, String name, String value) throws SecretManagementException {
+    public void deleteSecret(String secretType, String secretId)
+            throws SecretManagementException {
```
Comment (pulasthi7) [inline components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManagerImpl.java:163]: should fit 1 line

File: components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManagerImpl.java [hunk 3]
```diff
@@ -87,261 +79,182 @@ public Secret addSecret(String secretTypeName, Secret secret) throws SecretManag
...
-            log.debug("Secret type: " + secretType.getName() + " retrieved successfully.");
+            log.debug(secret.getSecretId() + " secret updated successfully.");
         }
-        return secretType;
+        return secret;
     }
 
     @Override
-    public void deleteSecretType(String secretTypeName) throws SecretManagementException {
+    public Secret updateSecretValue(String secretType, String secretId, String value)
```
Comment (pulasthi7) [inline components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManagerImpl.java:202]: do we need a dedicated method for this? update secret should be able to achieve this

File: components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManagerImpl.java [hunk 4]
```diff
@@ -487,117 +337,70 @@ private String encrypt(String plainText) throws CryptoException {
...
-    private void validateSecretTypeReplaceRequest(SecretType secretType) throws SecretManagementException {
-
-        if (StringUtils.isEmpty(secretType.getName())) {
-            throw handleClientException(ERROR_CODE_SECRET_TYPE_NAME_REQUIRED, null);
-        }
-
-        if (!isSecretTypeExists(secretType.getName())) {
+    private void validateSecretDescription(String description)
+            throws SecretManagementClientException {
+        if (description != null && !isSecretDescriptionRegexValid(description)) {
```
Comment (pulasthi7) [inline components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManagerImpl.java:397]: is discription madatory?

File: components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManagerImpl.java (hunk 3 above)
Comment (ThaminduDilshan) [reply components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManagerImpl.java:202]: This method is needed for internal operations. Internal components should be able to change secret value without retrieving the secret first.
@senthalan WDYT?

File: components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManagerImpl.java (hunk 4 above)
Comment (ThaminduDilshan) [reply components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManagerImpl.java:397]: No. This is the related regex, `"^.{0,1023}$"` which currently only checks for the length

File: components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManagerImpl.java (hunk 3 above)
Comment (senthalan) [reply components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManagerImpl.java:202]: yes. the internal components which use the secret management(IDP secret) should be able to update the secret value without pulling up the secret object from the database. The key of the secret will be stored at the component level and during the update, those components will only update the value.

**Comment by ThaminduDilshan** [comment]
Related FE changes are still not done and seems not planned.
Hence closing the PR

**Pull request closed without merging**